src/lib/y2nfs_client/nfs_version.rb

```diff
@@ -65,7 +65,8 @@ def self.all
     # @param value [String]
     # @return [NfsVersion]
     def self.for_mntops_value(value)
-      all.find { |version| version.mntops_value == value }
+      value = "4" if value == "4.0"
+      all.find { |version| version.mntops_value == value } or raise "Unknown mntops value #{value.inspect}"
```

Comment:
According to the documentation, this method is indeed not expected to return `nil`. But I found one example in which that seems to be the expectation of the caller.

https://github.com/yast/yast-nfs-client/blob/master/src/modules/Nfs.rb#L201

So I would:
 - raise a more specific exception type (maybe `ArgumentError`?),
 - update the documentation of `NfsVersion.for_mntops_value` to reflect the fact that it will raise an exception if no matching version is found,
- update the documentation of `Yast::NfsOptionsClass.nfs_version` in the same way
- and fix that line in `Yast::NfsClass` to use a rescue instead of that `version &&...`